app/jinja_filters.py

```diff
@@ -142,6 +152,45 @@ def format_conditional_date(context, date1=None, date2=None):
     return format_date(context, date)
 
 
+@blueprint.app_template_filter()
+def calculate_offset_date(input_datetime, offset, day_of_week='MO'):
```

Comment:
This is really hard to name but I think we may need to be clearer about the fact that this offset starts from a specified date in the last whole week. We'll need to buy a hat to put suggestions in.

Reply:
`calculate_date_offset_from_last_weekday` - I think that works?